org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdInventory.java

```diff
@@ -96,4 +128,31 @@ public String getTypeName(IId id) {
   public Class<? extends IId> getIdClass(String typeName) {
     return m_nameToClassMap.get(typeName);
   }
+
+  /**
+   * @return Whether the {@link IId} needs to be serialized and deserialized using encryption or not. For more details
+   *         see {@link IdEncryption}.
+   */
+  public boolean isIdEncryption(Class<? extends IId> idClass) {
+    synchronized (m_classToIdEncryptionMap) {
+      if (m_classToIdEncryptionMap.containsKey(idClass)) {
+        return m_classToIdEncryptionMap.get(idClass);
+      }
+
+      m_classToIdEncryptionMap.put(idClass, Optional.ofNullable(idClass.getAnnotation(IdEncryption.class))
+          .map(IdEncryption::value)
+          // check all super class
+          .orElseGet(() -> Optional.of(idClass.getSuperclass())
+              // only consider those that extend/implement IId
+              .filter(Objects::nonNull)
+              .filter(IId.class::isAssignableFrom)
+              .<Class<? extends IId>> map(c -> c.asSubclass(IId.class))
+              // check isIdEncryption
+              .map(this::isIdEncryption)
+              // true if optional is empty
+              .orElse(true)));
```

Comment:
why is the defaut (= add signature) hard-coded here?
the exclusions are marked in AbstractUuId and AbstractStringId -> why not mark AbstractRootId or AbstractLongId as "needs sign" and remove this default value (e.g. change to false (= no annotation))

Reply:
Annotation added to `IId` and used as fallback if no annotation is present on the super class hierarchy.